Child Ruto Actions Part 3: Boss room

JordanLongstaff opened the pull request:
Actions in the boss room, namely involving the blue warp.

Dragorn421 reviewed (commented):
I think that ruto actions may benefit from the seldom used naming pattern `ActorName_GroupName_ActionName`

For examples, see `BgBdanObjects_OctoPlatform_PauseBeforeDescending`, `BossGanon_LightBall_Update`, `DemoDu_CsGoronsRuby_AdvanceTo03` and co

so with this group of "boss room" actions, use `EnRu1_BossRoom_` as a common prefix

thoughts?

JordanLongstaff commented:
> Also it would help if you introduced an enum for action indices (indices into `sActionFuncs`) and renamed the enum members as you document the actions. Then we can see which actions link to which

I was gonna do that last, but I've added a PR now. Merge that one before any of these other ones.
